src/components/NFTPage/EventList/index.vue

```diff
@@ -30,17 +30,22 @@
 export default {
   name: 'EventList',
   props: {
-    nftHistory: {
+    items: {
       type: Array,
       default: undefined,
     },
-    populatedList: {
-      type: Array,
-      default: undefined,
+  },
+  computed: {
+    trimmedItems() {
+      if (this.items.length >= 10) {
+        return this.items.filter((id, index) => index <= 9).map(event => ({
+          ...event,
+        }));
+      }
+      return undefined;
     },
```

Comment:
Suggest to add a prop `trimmedCount` with default value `10`

return the list with `Array.slice()` e.g.`items.slice(0, trimmedCount)`
no need to return `undefined`